manticore/platforms/linux.py

```diff
@@ -151,6 +151,78 @@ def sync(self):
         """
         return
 
+class ProcSelfMaps(File):
+    def __init__(self, flags, linux):
+        self.filename = "_proc_self_maps_"
+        with open(self.filename, 'w') as mapsFile:
+            print(linux.current.memory.__proc_self__, file = mapsFile)
+        mode = mode_from_flags(flags)
+        if (mode != "rb"):
+            raise EnvironmentError("/proc/self/maps is only supported in read only mode")
+        self.file = open(self.filename, mode)
+
+    def __getstate__(self):
+        state = {"name": self.name, "mode": self.mode, "closed": self.closed}
+        state["pos"] = None if self.closed else self.tell()
+        return state
+
+    def __setstate__(self, state):
+        name = state["name"]
+        mode = state["mode"]
+        closed = state["closed"]
```

Comment:
I guess you can calculate `closed` given `pos`?
If used then closed could be a property? (One of those @property getter things)

Reply:
Yes I should be able to calculate closed with pos